code/StringTagField.php

```diff
@@ -0,0 +1,283 @@
+
+		return $this;
+	}
+
+	/**
+	 * @return int
+	 */
+	public function getLazyLoadItemLimit() {
+		return $this->lazyLoadItemLimit;
+	}
+
+	/**
+	 * @param int $lazyLoadItemLimit
+	 *
+	 * @return static
+	 */
+	public function setLazyLoadItemLimit($lazyLoadItemLimit) {
+		$this->lazyLoadItemLimit = $lazyLoadItemLimit;
+
+		return $this;
+	}
+
+	/**
+	 * @return null|DataObject
+	 */
+	public function getDataObject() {
+		if($this->dataObject) {
+			return $this->dataObject;
+		}
+
+		if($form = $this->getForm()) {
+			return $form->getRecord();
+		}
+
+		return null;
+	}
+
+	/**
+	 * @param DataObject $dataObject
+	 *
+	 * @return $this
+	 */
+	public function setDataObject(DataObject $dataObject) {
+		$this->dataObject = $dataObject;
+
+		return $this;
+	}
+
+	/**
+	 * {@inheritdoc}
+	 */
+	public function Field($properties = array()) {
+		Requirements::css(TAG_FIELD_DIR . '/css/select2.min.css');
+		Requirements::css(TAG_FIELD_DIR . '/css/TagField.css');
+
+		Requirements::javascript(TAG_FIELD_DIR . '/js/TagField.js');
+		Requirements::javascript(TAG_FIELD_DIR . '/js/select2.js');
+		Requirements::javascript(THIRDPARTY_DIR . '/jquery/jquery.js');
+		Requirements::javascript(THIRDPARTY_DIR . '/jquery-entwine/dist/jquery.entwine-dist.js');
+
+		$this->addExtraClass('silverstripe-tag-field');
+
+		$this->setAttribute('multiple', 'multiple');
+
+		if($this->getShouldLazyLoad()) {
+			$this->setAttribute('data-silverstripe-tag-field-suggest-url', $this->getSuggestURL());
+		} else {
+			$properties = array_merge($properties, array(
+				'Options' => $this->getOptions()
+			));
+		}
+
+		return $this
+			->customise($properties)
+			->renderWith(array("templates/StringTagField"));
+	}
+
+	/**
+	 * @return string
+	 */
+	protected function getSuggestURL() {
+		return Controller::join_links($this->Link(), 'suggest');
+	}
+
+	/**
+	 * @return ArrayList
+	 */
+	protected function getOptions() {
+		$options = ArrayList::create();
+
+		$source = $this->getSource();
+
+		if($source instanceof Iterator) {
+			$source = iterator_to_array($source);
+		}
+
+		$values = $this->Value();
+
+		foreach($source as $key => $value) {
+			$options->push(
+				ArrayData::create(array(
+					'Title' => $value,
+					'Value' => $key,
+					'Selected' => in_array($key, $values),
+				))
+			);
+		}
+
+		return $options;
+	}
+
+	/**
+	 * {@inheritdoc}
+	 */
+	public function setValue($value, $source = null) {
+		if($source instanceof DataObject) {
+			$name = $this->getName();
+			$value = $source->$name;
+		}
+
+		if($source instanceof SS_List) {
+			$value = $source->column('ID');
+		}
+
+		return parent::setValue($value);
+	}
+
+	/**
+	 * {@inheritdoc}
+	 */
+	public function getAttributes() {
+		return array_merge(
+			parent::getAttributes(),
+			array('name' => $this->getName() . '[]')
+		);
+	}
+
+	/**
+	 * {@inheritdoc}
+	 */
+	public function saveInto(DataObjectInterface $record) {
+		parent::saveInto($record);
+
+		if(!$record instanceof DataObject) {
```

Comment:
I actually had that backwards, sorry. :)

In any case, we should allow any object that implements that interface.